### PR TITLE
Adjust supported compilers statement in F18 readme

### PR DESCRIPTION
**Pull request by CarolineConcatto**
This patch is an update of the readme according to compilations done and also information given in flang dev mailing list.
 [F18] Which compilers configurations f18 have being tested/built?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,15 +53,23 @@ git clone https://github.com/flang-compiler/f18.git
...
-F18 is written in C++17.
+F18 is written in C++17 and it has LLVM libraries dependency.
 
 The code has been compiled and tested with
-GCC versions 7.2.0, 7.3.0, 8.1.0, and 8.2.0.
+GCC versions from 7.2.0 to 9.3.0.
 
 The code has been compiled and tested with
-clang version 7.0 and 8.0
+clang version 7.0, 8.0, 9.0 and 10
```
Comment (sscalpone) [inline README.md]: "10" is not spelled "10.0" -- minor issue.

File: README.md [hunk 2]
```diff
@@ -53,15 +53,23 @@ git clone https://github.com/flang-compiler/f18.git
...
 using either GNU's libstdc++ or LLVM's libc++.
 
+The code has been compiled on
+AArch_64, X86_64 and PPC64 servers
```
Comment (sscalpone) [inline README.md]: What are the correct spellings?  Maybe aarch64, x86_64, and ppc64le, but we should model the llvm targets.  I don't know the answer.

File: README.md [hunk 3]
```diff
@@ -53,15 +53,23 @@ git clone https://github.com/flang-compiler/f18.git
...
+with CentOS7, Ubuntu18.04, Rhel, MacOs, Mojave, XCode and
+Apple LLVM version 10.0.1.
+
+The code does not compile with Windows and
```
Comment (sscalpone) [inline README.md]: Does the code compile with any compilers on Windows?

Comment (isuruf) [reply README.md]: Not yet. After the 3 PRs https://github.com/flang-compiler/f18/pulls/isuruf are merged, 92% can be compiled. Remaining issue is the two `fork()` calls.

File: README.md (hunk 2 above)
Comment (RichBarton-Arm) [reply README.md]: AArch64 is the correct spelling for that one.

I think Steve's suggestion for intel and power is fine.

Comment (CarolineConcatto) [reply README.md]: Done!

File: README.md (hunk 1 above)
Comment (CarolineConcatto) [reply README.md]: Done!

**Comment by CarolineConcatto** [comment]
@sscalpone is this PR fine for us to merge?

**Comment by sscalpone** [comment]
@CarolineConcatto Please squash.

**Comment by CarolineConcatto** [comment]
@sscalpone  Done!

**Pull request merged**